src/components/pages/dashboard/profile.js

```diff
@@ -1,10 +1,202 @@
+
+  const [checked, setchecked] = useState(false);
+
+  const handleToggleChange = () => {
+    setchecked(!checked);
+  };
+
+  const handleAccountNameChange = (event) => {
+    setAccountName(event.target.value);
+  };
+  const handleCurrentPinChange = (event) => {
+    SetCurrentPinCode(event.target.value);
+  };
+  const handleNewPinChange = (event) => {
+    setNewPinCode(event.target.value);
+  };
+  const handleConfirmNewPineChange = (event) => {
+    setConfirmNewPinCode(event.target.value);
+  };
+
+  const closeUserAccount = () => {
+    const userIndex = getUserIndexByAccountNumber(currentUserAccountNumber);
+    registeredUsers.splice(userIndex, 1);
+    const registeredUsersLeft = JSON.stringify(registeredUsers);
+    localStorage.setItem("MB_USER_ACCOUNTS", registeredUsersLeft);
+    navigate("/");
+  };
+
+  const schema = yup.object({
+    accountName: yup.string().required("Account Name Required"),
+    accountNumber: yup.string(),
+    currentPin: yup.string(),
+    newPin: yup.string().length(4, "Account PIN must be exactly 4 digits"),
+    confirmNewPin: yup
+      .string()
+      .oneOf([yup.ref("newPin"), null], "Pins do not match"),
+  });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(schema),
+  });
+  const onSubmit = (data) => {
+    const currentUserIndex = getUserIndexByAccountNumber(
+      currentUserAccountNumber
+    );
+    registeredUsers[currentUserIndex].accountName = data.accountName;
+    if (checked === true && data.currentPin === "") {
+      alert("Insert Pin");
+      return;
+    }
+    if (
+      checked === true &&
+      data.currentPin !== registeredUsers[currentUserIndex].accountPin
+    ) {
+      alert("Pin Incorrect");
+      return;
+    }
+
+    if (checked === true && data.newPin !== data.confirmNewPin) {
+      alert("Pins do not match");
+      return;
+    }
+    if (
+      checked === true &&
+      data.confirmNewPin === registeredUsers[currentUserIndex].accountPin
+    ) {
+      alert("use a different password for update");
+    }
+    if (checked === true && data.newPin === "") {
+      data.newPin = registeredUsers[currentUserIndex].accountPin;
+    }
+    if (checked === true && data.newPin === data.confirmNewPin) {
+      registeredUsers[currentUserIndex].accountPin = data.newPin;
+    }
+
+    localStorage.setItem("MB_USER_ACCOUNTS", JSON.stringify(registeredUsers));
+    navigate("/transactions");
+  };
   return (
     <DashboardLayout pageTitle="Profile">
-      <p>This is my profile page</p>
+      <div className="container-fixed">
+        <form className="account-profile" onSubmit={handleSubmit(onSubmit)}>
+          <div className="form-group">
+            <label className="form-control-label">Account Name</label>
+            <input
+              type="text"
+              placeholder={accountName}
+              className="form-control"
+              onChange={handleAccountNameChange}
```

Comment:
This is not needed. useForm hook is registering and handling the input changes.

Reply:
Resolved